CMakeLists.txt

```diff
@@ -461,6 +461,23 @@ function(construct_add_custom_command_for_linker_pass linker_output_name output_
     set(linker_pass_define -DLINKER_PASS2)
   elseif (${linker_output_name} MATCHES "^linker_app_smem_unaligned$")
     set(linker_pass_define -DLINKER_APP_SMEM_UNALIGNED)
+  elseif (${linker_output_name} MATCHES "^linker$")
+    if(EXISTS ${APPLICATION_SOURCE_DIR}/pm.yml)
+      zephyr_compile_definitions(USE_PARTITION_MANAGER)
+      # Set a global property, so that all images (including the root image),
+      # get the variable set.
+      set_property(GLOBAL PROPERTY PROPERTY_PARTITION_MANAGER_TARGET
+        PARTITION_MANAGER_TARGET)
+      # Duplicates are removed before this list is used.
+      set_property(GLOBAL APPEND PROPERTY
+        ${IMAGE}PROPERTY_LINKER_SCRIPT_DEFINES
+        -DUSE_PARTITION_MANAGER)
+      # Below is needed to ensure that the origin app is placed correctly.
+      set_property(GLOBAL APPEND PROPERTY
+        PROPERTY_LINKER_SCRIPT_DEFINES
+        -DUSE_PARTITION_MANAGER)
```

Comment:
What's going on here?

won't IMAGE evaluate to '' when the app is processed and make the second invocation redundant?